src/Jackalope/NodeType/NodeType.php

```diff
@@ -182,7 +182,11 @@ public function canSetProperty($propertyName, $value, $throw = false)
     {
         $propDefs = $this->getPropertyDefinitions();
         try {
-            $type = $this->valueConverter->determineType(is_array($value) ? reset($value) : $value);
+            if (is_array($value) && 0 === count($value)) {
+                $type = PropertyType::STRING;
```

Comment:
i am not sure about this one. but when the array is empty and the type not specified, we need to chose something arbitrarily as there is no value to decide upon...